pkg/v1/tkg/client/machine_deployment.go

```diff
@@ -172,9 +172,7 @@ func DoSetMachineDeployment(clusterClient clusterclient.Client, options *SetMach
 		options.Name = fmt.Sprintf("%s-%s", options.ClusterName, options.Name)
 		machineTemplateName := fmt.Sprintf("%s-mt", options.Name)
 		kcTemplate.Name = fmt.Sprintf("%s-kct", options.Name)
-		if kcTemplate.Spec.Template.Spec.Files != nil && len(kcTemplate.Spec.Template.Spec.Files) > 0 {
```

Comment:
Preexisting: should such manipulation logic be done after checking for the right InfrastructureRef.Kind?